pkg/validation/port.go

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package validation
+
+import (
+	"fmt"
+
+	"github.com/projectriff/cli/pkg/cli"
+)
+
+func PortNumber(port int32, field string) cli.FieldErrors {
+	errs := cli.FieldErrors{}
+
+	if port < 0 || port > 65535 {
```

Comment:
```suggestion
	if port < 1 || port > 65535 {
```

Reply:
this is not called for a 0 value, so this change doesn't change any behavior

Reply:
"Programmers who specify "port 0" know that it is an invalid port. They are asking the operating system to pick and assign whatever non-zero port is available and appropriate for their purpose." from https://www.grc.com/port_0.htm

Reply:
that's exactly why I want it to be enforced. `0` is not a valid port, so the validator should not treat it as a valid port.

These validation methods may be used in other contexts in the future. We should implement a pure validator without assumptions about where it's called from.